Windows install instructions update for 2020

joelostblom opened the pull request:
I will make a few comments and then ping when this is ready for review.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+> Note that if you want to run Python directly from the command line on Windows, you need to append the `-i` flag, so the full command would be `python -i`. Running just `python` will freeze the Git Bash terminal.
 
-![](/resources_pages/imgs/jupyter_lab_r_kernel.png)
+One annoyance with our current terminal setup is that the word `(base)` is not on the same row as the rest of the prompt string (the part with `your_name@your_computer`. To fix this we can edit the `.bash_profile` configuration file to indicate that we do not want a newline at the beginning of the prompt string. Open up the configuration file using VS Code by typing the following command into a t ...
```
This is a bug, which there is an issue about on the conda installer.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-To be able to call R scripts from the Powershell or Git Bash you need to be able to execute the Rscript program from there. To make this possible you need to add the the file path of the directory/folder that houses the Rscript program (i.e., location where this program lives on your computer) to the Windows PATH system variable. You can modify the PATH system variable using System Utility in con ...
+Click to save the file,
+when VS Code prompts you that the saving failed,
+click "Retry as Admin" and then "Yes".
```
This is a really nice function of vscode, never have to right click and "open as admin"!

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
 
-![](/resources_pages/imgs/RStudio.png)
+```
+conda init bash
````
And I added to PATH manually instead of recommending to use the anaconda prompt, since I don't think it will be a good experience keeping track of two different terminals.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+conda install nodejs=10.*
+pip install --upgrade jupyterlab-git
+conda install jupytext=1.*
+jupyter lab build
```
I think jupyter will build automatically next time it is open, but might as well have it here I guess.

ttimbers replied:
I think the `jupyter lab build` should be done here. Otherwise the user will be prompted to build when they open Jupyter, which I find annoying and new students might find confusing.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-2213894ea0415ee8c85c5eea0d0ff81ecc191529
-x64
+# Add R and Rscript to PATH
+export PATH="/c/Program Files/R/R-4.0.2/bin/x64":$PATH
```
I don't know if they will be using R from terminal other than below for the setup, but thought it would make it sense to enable it so that students can see how it is done and use if needed.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
+By default, RStudio installs packages to `C:/Users/your_username/Documents/R/win-library/4.0`, but when it is run directly from the terminal it instead uses  `C:/Users/your_username/R/win-library/4.0`. To make sure that R from terminal uses the same package directory as R studio, we will edit `~/.Rprofile` by typing:
+
+```
+code ~/.Rprofile
````
The amount of hours it took me to figure out where to put what here shall remain unnamed. Do you think this is the best solution? I found so many different solutions here, including editing `R_LIBS_PATH`, .Rprofile at two places, .Renviron at two places. The one I suggested worked the most robustly of the ones I tried and is also the most alike how they are editing e.g. `.bash_profile` so I thought what they would learn there would transfer to here (and this also serve as a way of solidifying this concept of editing profile files.

ttimbers replied:
We should be setting `R_LIBS` in `.Renviron` not `.Rprofile` as suggested here: https://github.ubc.ca/MDS-2019-20/DSCI_522_dsci-workflows_students/issues/4

In general, these two main files serve these two functions: 
- `.Renviron` - which contains environment variables to be set in R sessions.
- `.Rprofile` - which contains R code to be run in each session.

(from @jennybc's [What They Forgot to Teach You About R](https://rstats.wtf/index.html) book). More can be found on these two files in this chapter: https://rstats.wtf/r-startup.html#renviron

joelostblom replied:
Great, thank you! I will update this.

joelostblom replied:
One more thing I realized as I was going to make this change on my machine. The [Rtools website recommend doing the PATH manipulations in `.Rprofile`](https://cran.r-project.org/bin/windows/Rtools/), do you think it will be confusing for students that we are modifying env paths in two different files? And is PATH a special case that cannot be set in `.Renviron`, or are they technically doing modifying the wrong file?

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
+by typing the following into the R terminal in RStudio:
+
+```
+install.packages(c('tidyverse', 'blogdown', 'xaringan', 'renv'))
````
There were lots of more packages in the orignal windows install, I changed to jsut the ones from Ububntu and Mac, but if we need some of the other ones, let me know and I will add them back.

joelostblom replied:
These were the original ones 

```
install.packages(c('repr', 'IRdisplay', 'evaluate', 'crayon', 'pbdZMQ', 'devtools', 'uuid', 'digest', 'tidyverse'))
```

ttimbers replied:
Ahhh, I think this is the order in which we have changed the install instructions... 'repr', 'IRdisplay', 'evaluate', 'crayon', 'pbdZMQ', 'uuid', and 'digest'' are all dependencies of the `IRKernel` package... (see: https://github.com/IRkernel/IRkernel/blob/9871fc1aafc7b8da34b81c8511606d596b9aa99c/DESCRIPTION#L25).

ttimbers replied:
So I don't think they are needed, they should be installed later as dependencies when the  `IRKernel` package is installed.

joelostblom replied:
Good to know, I didn't run into any problems with IRkernel not finding packages when running from Jupyter, so if these are not used foranything else we should be good

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+
+### Rtools
+
+Windows users will also need to install Rtools, which will allow you to use external libraries. Go to <http://cran.r-project.org/bin/windows/Rtools/> and download the latest version (e.g., Rtools40.exe). After the download has finished, run the installer with the default configuration. After the installation has completed, follow the steps on the Rtools page under the heading "Putting Rtools on t ...
```
I changed this to refer to the setup and verification instructions on the webpage instead of duplicating them here.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+
+### IR kernel
+
+The `IRkernel` package is needed to make R work in Jupyter notebooks. To enable this kernel in the notebooks, open R *from a terminal* and run the setup via the following two commands:
```
I tried for a few hours to get RStudio to see my jupyter installation but no luck. Even when I added it to path the same way as I added rtools above, it would say it found Jupyter but could not run it. Launching from Git Bash sets the R PATH to the path setu in gitbash which contains all miniconda executables. This is now similar to the macOS instructions which also uses R in terminal for this step.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-*note - if a screen appears titled "Windows protected your PC", click on "More info" link and the "Run anyway" button*
+First, open RStudio and run the following commands to install the `tinytex` package and setup `tinytex`:
 
-After installation, restart Powershell and type the following to ask for the version:
```
I removed all mentions of powershell to make it more unix like

ttimbers replied:
THANK-YOU!

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
+tinytex::install_tinytex()
+```
+
+Note that you might see two error messages regarding lua during the installation, you can safely ignore these, the installation will complete successfully after clicking "OK".
````
This is strange, literal error messages saying things failed, but it installs fine, also noted in the tinytex installation instructions so it will likely be widespread.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
+by pasting the following:
+
 ```
+tlmgr.bat install eurosym \
````
we need .bat here

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+
+The above is all we need to have LaTeX work with R Markdown documents, however for Jupyter we need to add several more packages. Before we do this, please sign out of Windows and back in again in order for the Git Bash terminal to be able to find the location of TinyTex.
+
+When you sign back in,
```
There might be a way to add it manually, but I like that it happens automatically when signing in and out.

ttimbers replied:
This seems fine.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+If the PDF file is created,
+your LaTeX environment is set up correctly.
+
+> Note if you see an error that says "Undefined control sequence", you need to edit a configuration file for Jupyter's PDF export. Open a terminal and paste `code "/c/Users/${USERNAME}/miniconda3/Lib/site-packages/nbconvert/exporters/pdf.py"`. On line 66, it says `writer = Instance("nbconvert.writers.FilesWriter", args=(), kw={'build_directory': '.'})`, replace this line with the following line ` ...
```
This is very annoying and could be confusing for students, I will ask if they accept a simple PR but from the issue, it seems like they are reluctant or think it could introduce other ptorblems since it has been open since last year.

joelostblom replied:
@ttimbers I suggested a fix in this PR https://github.com/jupyter/nbconvert/pull/1345

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-We will be using Make to automate our analysis scripts. More on this later!
 
-Go to https://github.com/gnu-mcu-eclipse/windows-build-tools/releases and download `gnu-mcu-eclipse-windows-build-tools-2.12-20190422-1053-win64.zip`. 
+Later in the program, we will be using `make` to automate our analysis scripts. [Download `make` from this URL](https://downloads.sourceforge.net/project/ezwinports/make-4.3-without-guile-w32-bin.zip). Click on the downloaded zip-file to open it in the File Explorer, click the "Extract" tab and the button in the header that says "Extract all" and accept the default extraction location.
```
I simplified this and also took the same exe that the scoop package manager uses and I sawa recommended elsewhere

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
 
-Next, like you did when installing [Rscript](https://github.com/UBC-MDS/UBC-MDS.github.io/blob/master/resources_pages/install_ds_stack_windows.md#rscript), you need to add the folder where `make.exe` is located to the _PATH_ variable. The `make.exe` file is located in C:\Users\\`user_name`\AppData\Roaming\GNU MCU Eclipse\Build Tools\2.12-20190422-1053\bin.
+```
+code ~/.bash_profile
````
Again hammering home this concept.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+
+We will be using PostgreSQL as our database management system. You can [download PostgreSQL from here](https://www.enterprisedb.com/downloads/postgres-postgresql-downloads). Follow the instructions for the installation. In the password page, type whatever password you want, but make sure you'll remember it later. For all the other options, use the default. You do not need to run "StackBuilder" at ...
+
+To test if the installation was successful open the `SQL Shell` app from the Start menu. You will be asked to setup your configuration, accept the default value (the one within square brackets) for the first four values by pressing enter four times, then type in your password and press enter one last time. It should look like this if it is working correctly:
```
Make them run through the config so they understand what it is and practice remembering their passworkd

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
+ https://docs.docker.com/get-started/
 ```
 
+### Customizing VS Code with Extensions
````
Should we move this to the vscode section? It is not needed for anything else I think so breaks the flow, but also clearly belons to vscode.

ttimbers replied:
I initially had it higher up, but these extensions work with things like Git, Python and Docker, which are installed later than VS Code, so that we can use VS Code as an editor when installing and configuring things. Therefore I moved them to the end and believe they should stay there.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
+For MDS program announcements, course forums, and correspondence we use the communication tool Slack. Slack can be accessed via the web browser, but we recommend using the Slack app, which can be installed via downloadable file from the slack website [https://slack.com/intl/en-ca/downloads/windows](https://slack.com/intl/en-ca/downloads/windows).
+
+
+## Visual Studio Code
```
Should we have this same setup on all OSes? vscode can also be set as the diff tool, but I think that might be too far for now, editor might make sense across OSes though...

ttimbers replied:
We should have the same editor across all OS's to make teaching easier. If you plan on using VS Code and spending a lot of time on local diffs then yes, you might want to make sure all 3 OS's use this for that then. I like to diff on GitHub personally, however, I know sometimes it is useful to do this stuff locally (I just tend not to do this very much myself).

joelostblom replied:
I updated this to set EDITOR and VISUAL in all OSes so that vscode is the default editor for everything.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-![](/resources_pages/imgs/jupyter_lab_r_kernel2.png)
+Delete the line that reads the following (it should be line 13):
 
-### Rscript
```
removed because we add all R executables to path now.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,543 @@ layout: page
-  - On the **Select Components** page, check "On the Desktop" under "Additional icons"
-  - On the **Choosing the default editor used by Git** page, select "Use the Nano editor by default" from the drop-down menu
-  - For all other pages, use the default options
+> Note that we don't recommend pinning Git Bash to the taskbar since launches Git Bash from here will set a different home directory than if Git Bash is launched from the desktop or the start menu (`/` instead of `/c/users/$USERNAME`.
```
Things like this makes me want to force everyone to dual boot Ubuntu...

joelostblom replied:
@ttimbers I found the solution for this here https://github.com/git-for-windows/git/issues/1790#issuecomment-412493650

> That happens because Git GUI is a script and when you try to pin a running script to the taskbar, windows pins the executable that runs the script instead. Navigate to Git GUI in your start menu, right click the entry and select "pin to taskbar" to actually pin Git GUI (well, the GfW wrapper that launches Git GUI) to your taskbar.

Will update this note.

joelostblom commented:
@ttimbers Ready for review, there is quite a lot in this one. I tried to make it more unix like overall as I think this is better knowledge for students to have and also closer follows recommendations such as not adding anaconda to the global path. What do you think?

I think there is still a lot of "click here to download X and then do Y", which could be confusing or tiring when there are so many programs to install. In the future a windows package manager such as chocolatey, or maybe mysys or pacman could be good (although wsl might make this obsolete).

ttimbers commented on resources_pages/install_ds_stack_mac.md:
````diff
@@ -301,6 +303,29 @@ install.packages('tinytex')
+You can check that the installation is working by typing the following in a terminal to ask for the version of latex:
+
+```
+latex --version
````
Sure

ttimbers commented on resources_pages/install_ds_stack_mac.md:
```diff
@@ -331,36 +356,13 @@ Go to `File -> Export notebook as... -> Export Notebook to PDF`.
+We will be using PostgreSQL as our database management system. You can [download PostgreSQL from [here](https://www.enterprisedb.com/downloads/postgres-postgresql-downloads). Follow the instructions for the installation. In the password page, type whatever password you want, but make sure you'll remember it later. For all the other options, use the default. You do not need to run "StackBuilder" a ...
 
-To test if the installation was successful open the `SQL Shell` app from the LaunchPad or applications directory. It should look like this if it is working correctly: 
+To test if the installation was successful open the `SQL Shell` app from the LaunchPad or applications directory. You will be asked to setup your configuration, accept the default value (the one within square brackets) for the first four values by pressing enter four times, then type in your password and press enter one last time. It should look like this if it is working correctly:
```
Sounds good.

ttimbers commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,540 @@ layout: page
-  - On the **Choosing the default editor used by Git** page, select "Use the Nano editor by default" from the drop-down menu
-  - For all other pages, use the default options
+Go to <https://git-scm.com/download/win> and download the windows version of git. After the download has finished, run the installer and accept the default configuration for all pages except for the following:
+  - On the **Choosing the default editor used by Git** page, select "Use Visual Studio Code as Git's default editor" from the drop-down menu'
```
I don't think they will ever need nano. I do think they should learn the basics of vim (most importantly exiting vim 😉 , followed by editing a file) as this is important when you get dumped on a bare server. However, in reality this happens usually in the cloud/server context and so we teach it there.

ttimbers commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,540 @@ layout: page
 
-We will be using Python for a large part of the program, including many popular 3rd party Python libraries for scientific computing. [__Anaconda__](https://www.anaconda.com) is an easy-to-install distribution of Python and most of these libraries (as well as Jupyter notebooks, one of the developing environments we will be using). We __strongly recommend__ that you use Anaconda for this program. I ...
+```
+bash --version
````
I don't think we heavily use bash enough that we need to update this. We have not yet had a bash problem on Mac before.

ttimbers commented:
OK, some comments made above from reviewing your comments. I will now work through these install instructions on my Windows machine and add more comments/suggestions as needed.

ttimbers commented:
@joelostblom - I have updated the R install instructions more in the right direction. Can you take a read over and massage them to be a bit more user friendly? They are at least correct now. This idea might help:

For these things, I use (which works on all platforms and after upgrading R):

- Set R_LIBS_USER in your ~/.Renviron. One <name>=<value> per line.
- Don't hardcode the R version or architecture. Instead make use of so-called "specifiers", which include %p (expands to the architecture, e.g. x86_64-pc-linux-gnu) and %v (expands to major and minor R version, e.g. 3.5) - see ?R_LIBS_USER. Using R_LIBS_USER=~/R/%p-library/%v corresponds to the default R settings.

source: https://community.rstudio.com/t/help-regarding-package-installation-renviron-rprofile-r-libs-r-libs-site-and-r-libs-user-oh-my/13888/8

Also, I think we might want to add `usethis` and `devtools` to our base R install. I added for windows, can you please add for Ubuntu and Mac?

joelostblom commented:
Thanks for reviewing @ttimbers !

> [ ]  I am not sure about the modifying the PS1 variable to move `(base)` to the same line is worth it. I think students might find this annoying and pedantic... Can we instead open an issue in the GitBash or conda repo (or somewhere else?) and work with the dev team upstream to fix this?

What do you think of turning this into an optional quoted note section? Then people who want to fix don't have to waste time looking it up, and students who don't care can just eave it as is.

This is the issue https://github.com/conda/conda/issues/10031, there has not been any responses. I don't know what a good solution would be, check for git bash installation (or windows OS) + newline in the beginning of the prompt? I don't know if it is possible for conda to modify PS1 before `git-prompt.sh` does it (and I think it would be reset anyways). The ideal would maybe be if the ps1 config from was all in `bash_profile` I don't think that the Git bash maintainer would be interested in such a big change to acommodate conda. 

> [ ]  In a couple places they have to choose between 32 and 64 bit... To newbies this might be a confusion point. Can we add some direction here?

Will do, I will click through the links and check which had this choice.
 
> [ ]  Under "Essential Python packages", or before in the configuring Git Bash section, can we add instructions on how to copy and paste into Git Bash (Ctrl + V doesn't work, you need to right click and select paste).

Yes will do, I commented more on this in https://github.com/UBC-MDS/UBC-MDS.github.io/issues/70
 
> Under "Essential Python packages" (sorry, I can't seem to add an inline comment in the PR review for some of these, sigh...) without the `-y` flag for `conda install nodejs=10.*` it takes the next line of text `pip install ...` as the keyboard input roll_eyes ... So we either need to add the `-y` here or have them paste these in one line at a time... I opted for the `-y` option I paste below:
> 
> ```
> conda install -y nodejs=10.*
> pip install --upgrade jupyterlab-git
> conda install -y jupytext=1.*
> jupyter lab build
> ```
> 

Good point, I pasted those lines one by one without realizing that students might paste it all at once. I guess we can't have the pip line first because we need node for the jupyterlab-extension part of jupyterlab-git? I agree that adding `-y` is a good solution here. Btw, is there any reason we don't use a single call to conda, so `conda install -y nodejs=10.* jupytext=1.*`?

> In these instructions (and potentially the other OS's, anytime they edit `.Renviron` and/or `.Rprofile` (and likely `.bash_profile`) they need to leave a blank line at the end of the file. I got into silly and embarrassing trouble with this in `.Renviron` recently...

Hmmm, to me this sounds like the silliness is on the side of either the program or maybe the text editor for allowing such as hard to miss error to happen in the first place... I know vim adds newlines, and I found this about vscode having a similar option https://github.com/Microsoft/vscode/issues/12076 Not sure if it is worth mentioning.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,543 @@ layout: page
+## Visual Studio Code
+
+The open-source text editor Visual Studio Code (VS Code) is both a powerful text editor and a full-blown Python ADE, which we will use for more complex analysis. Go to [https://code.visualstudio.com/download](https://code.visualstudio.com/download) and download the windows version of VS Code. After the download has finished, run the installer and accept the default configuration for all pages exc ...
+    - On the **Select Additional Tasks** check all four boxes under "Other"
+        - "Add 'Open with Code' action to Windows file context menu"
+        - "Add 'Open with Code' action to Windows directory context menu"
+        - "Register Code as an editor for supported file types"
+        - "Add to PATH" (this should selected by default).
```
Will fix

joelostblom commented on resources_pages/install_ds_stack_windows.md:
```diff
@@ -3,290 +3,543 @@ layout: page
 
-We will be using the command line version of Git as well as Git through RStudio. 
+Go to <https://git-scm.com/download/win> and download the windows version of git. After the download has finished, run the installer and accept the default configuration for all pages except for the following:
+    - *Optional* On the **Select Components** page, check "On the Desktop" under "Additional icons".
+    - On the **Choosing the default editor used by Git** page, select "Use Visual Studio Code as Git's default editor" from the drop-down menu'
```
Will fix bullets and screenshots

joelostblom commented:
> @joelostblom - I have updated the R install instructions more in the right direction. Can you take a read over and massage them to be a bit more user friendly? They are at least correct now. This idea might help:
> 
> For these things, I use (which works on all platforms and after upgrading R):
> 
> - Set R_LIBS_USER in your ~/.Renviron. One = per line.
> - Don't hardcode the R version or architecture. Instead make use of so-called "specifiers", which include %p (expands to the architecture, e.g. x86_64-pc-linux-gnu) and %v (expands to major and minor R version, e.g. 3.5) - see ?R_LIBS_USER. Using R_LIBS_USER=~/R/%p-library/%v corresponds to the default R settings.
> 
> 
> source: https://community.rstudio.com/t/help-regarding-package-installation-renviron-rprofile-r-libs-r-libs-site-and-r-libs-user-oh-my/13888/8
> 
> Also, I think we might want to add `usethis` and `devtools` to our base R install. I added for windows, can you please add for Ubuntu and Mac?

I will masage this and incorporate your suggested updates later this week, maybe Friday.

ttimbers commented:
> What do you think of turning this into an optional quoted note section? Then people who want to fix don't have to waste time looking it up, and students who don't care can just eave it as is.

Sounds good to me.

> This is the issue conda/conda#10031, there has not been any responses. I don't know what a good solution would be, check for git bash installation (or windows OS) + newline in the beginning of the prompt? I don't know if it is possible for conda to modify PS1 before git-prompt.sh does it (and I think it would be reset anyways). The ideal would maybe be if the ps1 config from was all in bash_profile I don't think that the Git bash maintainer would be interested in such a big change to acommodate conda.

If we both post in that issue thread that we'd like to see that fixed maybe someone might help? Or at least suggest a starting point?

> Btw, is there any reason we don't use a single call to conda, so conda install -y nodejs=10.* jupytext=1.*

Not that I know of. But you should uninstall and test before you decide to change this.

> Hmmm, to me this sounds like the silliness is on the side of either the program or maybe the text editor for allowing such as hard to miss error to happen in the first place... I know vim adds newlines, and I found this about vscode having a similar option microsoft/vscode#12076 Not sure if it is worth mentioning.

Yeah, I ended up here by using `echo COMMAND > FILE` which doesn't use a newline... I just like to add the note incase someone else does something similar...

> I will masage this and incorporate your suggested updates later this week, maybe Friday.

Sounds good - check in with me before you do... I have a few ideas/improvements I might want to check/see...

joelostblom commented:
> If we both post in that issue thread that we'd like to see that fixed maybe someone might help? Or at least suggest a starting point?

I posted just now!

joelostblom commented:
@ttimbers  Ready for another review!

I am finalizing the script (I want to print path and their bash_profile to the log file), then we can change the URL once it is merged (pointing to my repo for now). I want to add a section with a bash_prfile/rc file that makes it a bit nicer (more like zsh). I will add that in a separate PR since this one already has plenty of scoop creep.

joelostblom commented on resources_pages/install_ds_stack_windows.md:
````diff
@@ -3,290 +3,632 @@ layout: page
+```
 
-We will be using Python for a large part of the program, including many popular 3rd party Python libraries for scientific computing. [__Anaconda__](https://www.anaconda.com) is an easy-to-install distribution of Python and most of these libraries (as well as Jupyter notebooks, one of the developing environments we will be using). We __strongly recommend__ that you use Anaconda for this program. I ...
+> If you tried to paste the above into the Git Bash terminal, you will have noticed that `Ctrl+V` does not work in Git Bash. Instead you need to right click and select "Paste" or use the `Shift+Insert` shortcut. To copy from the Git Bash terminal you simply select the text you want and it is copied automatically.
````
@ttimbers Could you confirm that this is the default setting for you? I think it is, but I am not 100% on whether I did change it and there is no "reset to default settings". I could also reinstall git bash and check later.

ttimbers replied:
I can't confirm as my Windows machine is at work and I can't get to it for another day or two. But I do remember students doing this, so let's go with it.

joelostblom commented on resources_pages/install_ds_stack_mac.md:
```diff
@@ -120,17 +125,38 @@ git version 2.24.2 (Apple Git-127)
-Next, we need to configure Git by telling it who you are, your email, and setting the default text editor to VS Code. To do this type the following into the terminal (replacing Jane Doe and janedoe@example.com, with your name and email, respectively):
+### Configuring Git user info
+
+Next, we need to configure Git by telling it your name and email. To do this type the following into the terminal (replacing Jane Doe and janedoe@example.com, with your name and email (the same used to sign up for GitHub), respectively):
```
Should they use their Github.com or ubc.github.ca email as the default?

ttimbers replied:
It doesn't matter, I don't think Git will know the difference, it just takes one email. TBH, I don't think it even has to be an email associated with the GitHub accounts. I think it's just a recommendation about it matching up to the one they use with GitHub. You can associate multiple emails with GitHub, and I think even associate them with different orgs.

joelostblom commented:
@ttimbers I made a minor update regarding `winpty`, this is ready for review.

ttimbers commented:
Looks good, let's go for it and see how it goes! Thanks for the major revisions @joelostblom - fingers crossed this effort results in smoother installs across the board!
